src/Microsoft.Graph.Cli.Core/IO/AuthenticationCacheUtility.cs

```diff
@@ -66,7 +77,82 @@ public async Task SaveAuthenticationIdentifiersAsync(string? clientId, string? t
         }
     }
 
-    public void DeleteAuthenticationIdentifiers()
+    public async Task<AuthenticationRecord?> ReadAuthenticationRecordAsync(CancellationToken cancellationToken = default)
+    {
+        var recordPath = Path.Combine(pathUtility.GetApplicationDataDirectory(), Constants.AuthRecordPath);
+        AuthenticationRecord? record = null;
+
+        if (File.Exists(recordPath))
+        {
+            using var authRecordStream = new FileStream(recordPath, FileMode.Open, FileAccess.Read);
+            var authRecord = await AuthenticationRecord.DeserializeAsync(authRecordStream, cancellationToken);
+            record = authRecord;
+        }
+
+        return record;
+    }
+    public async Task ClearTokenCache(CancellationToken cancellationToken = default)
+    {
+        // https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-net-clear-token-cache
+        // Work-around until https://github.com/Azure/azure-sdk-for-net/issues/32048 is resolved
+        var options = await ReadAuthenticationIdentifiersAsync(cancellationToken);
+        var record = await ReadAuthenticationRecordAsync(cancellationToken);
+        var clientId = record?.ClientId ?? options?.ClientId;
+        var tenantId = record?.TenantId ?? options?.TenantId;
+        if (options == null || string.IsNullOrWhiteSpace(clientId))
+        {
+            return;
+        }
+
+        var isPrivate = IsPrivateClient(options.Strategy);
+
+        IClientApplicationBase app;
+        var authority = options?.Authority?.Trim();
+        if (!string.IsNullOrEmpty(authority))
+        {
+            if (!authority.StartsWith("http"))
+            {
+                authority = $"https://{authority}";
+            }
+
+            authority = $"{authority}/{tenantId}";
```

Comment:
We can also probably delete line 110 to 119 as the `authority` isn't really used/referenced by the code.